### PR TITLE
LwM2M boostrap with different content formats

### PR DESCRIPTION
**Pull request by VeijoPesonen**
* Makes possible to use SenML CBOR and SenML JSON during bootstrap instead of TLV.
* Fixes SenML CBOR opaque data handling bug
* Does not report a missing optional resource as an error

### CODE REVIEW

**Comment by VeijoPesonen** [comment]
@jheiskan81 , @SeppoTakalo please review.

File: subsys/net/lib/lwm2m/lwm2m_rw_senml_cbor.c [hunk 1]
```diff
@@ -464,21 +466,21 @@ static int get_opaque(struct lwm2m_input_context *in,
...
+	} else {
+		LOG_DBG("Blockwise transfer not supported with SenML CBOR");
+		__ASSERT_NO_MSG(false);
 	}
 
-	return lwm2m_engine_get_opaque_more(in, value, buflen,
-					    opaque, last_block);
-not_supported:
-	return -ENOTSUP;
+	return dest ? opaque->len: -EINVAL;
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/lwm2m_rw_senml_cbor.c:483]:
```suggestion
	return dest ? opaque->len : -EINVAL;
```

**Review by rlubos — COMMENTED** [review]
Looks good, but aren't some commits missing the body part?

**Comment by VeijoPesonen** [comment]
> Looks good, but aren't some commits missing the body part?

You're right, should be fixed now.

File: subsys/net/lib/lwm2m/lwm2m_rw_senml_cbor.c (hunk 1 above)
Comment (VeijoPesonen) [reply subsys/net/lib/lwm2m/lwm2m_rw_senml_cbor.c]: Fixed.

**Review by SeppoTakalo — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**